aip/0194.md

```diff
@@ -0,0 +1,87 @@
+---
+aip:
+  id: 194
+  state: reviewing
+  created: 2019-08-16
+permalink: /194
+redirect_from:
+  - /0194
+---
+
+# Retries for gRPC clients
+
+RPCs sometimes fail. When one does, the client performing the RPC needs to know
+whether it is safe to retry the operation. When status codes are used
+consistently across multiple APIs, clients can respond to failures
+appropriately.
+
+## Guidance
+
+Clients **should** retry requests for which repeated runs would not cause
+unintended state changes, which are non-transactional, and which are either
+unary or server streaming.
+
+Clients **should not** retry transactional requests; instead these requests
+**should** have user-level retry logic that retries the entire transaction
+block from the start.
+
+Clients **should not** retry requests in which repeated runs would cause
+unintended state changes.
+
+**Note:** This AIP does not cover client streaming or bi-directional streaming.
+
+### Retryable codes
+
+The following codes **should** be retried for retryable requests:
+
+| Code | Discussion |
```

Comment:
Could we change back to lists? This ended up not looking as good as I hoped in GitHub Pages, and the Markdown also looks bad.

Reply:
Done